test/jdk/java/awt/Scrollbar/ScrollbarKeyControlTest.java

```diff
@@ -0,0 +1,169 @@
+                frame.add("East", scrollbarV);
+
+                frame.setSize(200, 200);
+                frame.setVisible(true);
+                frame.validate();
+                frame.toFront();
+            });
+            try {
+                robot = new Robot();
+                testOneScrollbar(scrollbarV);
+                if (changesTotal != 9) { //one by mouse click and six by keys
+                    throw new RuntimeException("Test failed.  Not all adjustment " +
+                            "events received by vertical scrollbar (" + changesTotal + " of 9)");
+                }
+                changesTotal = 0;
+                testOneScrollbar(scrollbarH);
+                if (changesTotal != 9) { //one by mouse click and six by keys
+                    throw new RuntimeException("Test failed.  Not all adjustment " +
+                            "events received by horizontal scrollbar (" + changesTotal + " of 9)");
+                }
+            } catch (AWTException e) {
+                throw new RuntimeException("Test interrupted.", e);
+            }
+            System.out.println("Test passed. Adjustment Event called  "
+                    + changesTotal + " times for each scrollbar");
+
+        } finally {
+            EventQueue.invokeAndWait(() -> {
+                if (frame != null) {
+                    frame.dispose();
+                }
+            });
+        }
+    }
+
+    public void testOneScrollbar(Scrollbar sb) {
+        robot.waitForIdle();
+        robot.mouseMove(sb.getLocationOnScreen().x + sb.getWidth() / 2,
+                sb.getLocationOnScreen().y + sb.getHeight() / 2);
+        try {
+            synchronized (LOCK) {
+                robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+                robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+                LOCK.wait(2000);
+                robot.keyPress(KeyEvent.VK_DOWN);
+                robot.keyRelease(KeyEvent.VK_DOWN);
+                LOCK.wait(2000);
+                robot.keyPress(KeyEvent.VK_PAGE_DOWN);
+                robot.keyRelease(KeyEvent.VK_PAGE_DOWN);
+                LOCK.wait(2000);
+                robot.keyPress(KeyEvent.VK_UP);
+                robot.keyRelease(KeyEvent.VK_UP);
+                LOCK.wait(2000);
+                robot.keyPress(KeyEvent.VK_PAGE_UP);
+                robot.keyRelease(KeyEvent.VK_PAGE_UP);
+                LOCK.wait(2000);
+                robot.keyPress(KeyEvent.VK_RIGHT);
+                robot.keyRelease(KeyEvent.VK_RIGHT);
+                LOCK.wait(2000);
+                robot.keyPress(KeyEvent.VK_LEFT);
+                robot.keyRelease(KeyEvent.VK_LEFT);
+                LOCK.wait(2000);
+                robot.keyPress(KeyEvent.VK_HOME);
+                robot.keyRelease(KeyEvent.VK_HOME);
+                LOCK.wait(2000);
+                robot.keyPress(KeyEvent.VK_END);
+                robot.keyRelease(KeyEvent.VK_END);
+                LOCK.wait(2000);
```

Comment:
@TejeshR13 Is a delay of 2s required between key presses? Seems a bit more than usual.

Reply:
This is not a delay, maximum wait time is 2sec which doesn't seem to happen during the test execution unless scrollbar adjustment is not invoked. I think max wait time of 2sec in case of unlock failure is fine.